test/python/pulse/test_block.py

```diff
@@ -429,6 +430,25 @@ def test_inherit_from(self):
         self.assertEqual(new_sched.name, ref_name)
         self.assertDictEqual(new_sched.metadata, ref_metadata)
 
+    @unittest.skipUnless(has_aer(), "qiskit-aer doesn't appear to be installed.")
+    def test_execute_block(self):
+        """Test executing a ScheduleBlock on a Pulse backend"""
+
+        ref_header = {"memory_slots": 1, "name": "test_block", "metadata": {}, "ode_t": 0.0}
+        ref_data = dict()
+        ref_data["counts"] = {"0x0": 1024}
+        ref_data["statevector"] = [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0]]
+
+        with pulse.build(name="test_block") as sched_block:
+            pulse.play(pulse.Constant(160, 1.0), pulse.DriveChannel(0))
+            pulse.acquire(50, pulse.MeasureChannel(0), pulse.MemorySlot(0))
+
+        test_job = self.backend.run(sched_block)
+        test_result = test_job.result().results[0]
+
+        self.assertEqual(test_result.header.to_dict(), ref_header)
+        self.assertEqual(test_result.data.to_dict(), ref_data)
```

Comment:
We don't need to do the whole result payload comparison here. It's mostly not a concern of the execute path and will depend on the qiskit-aer version (as that's what's being used under the covers for the simulation and what dictates the header contents). Instead of doing this I would delete L437-L440 and skip the whole `ref_data` dict creation and replace this assertion with something like:

```suggestion
        self.assertDictAlmostEqual(test_result.get_counts(), {'0': 1024}, 20)
```